### PR TITLE
Implement `MarsDataset` to integrate with PyTorch

### LINKED ISSUE #855: Support `run_pytorch_script` and `MarsDataset` or `MarsSampler` for pytorch
**Is your feature request related to a problem? Please describe.**

Now we have supported `run_tensorflow_script` to start distributed tensorflow cluster which introduced in #820 . We can provide `run_pytorch_script` as well, and since [PyTorch dataset API](https://pytorch.org/docs/stable/data.html) is more Pythonic ( just to implement `__getitem__` protocol and so forth), we can implement `MarsDataSet` or  `MarsSampler` to accelerate the preprocess for deep learning.

### PR DESCRIPTION
**Pull request by hekaisheng**
## What do these changes do?

- Add `MetaAPI` and `WorkerAPI`.

- Implement `MarsTorchDataset` which inherit `torch.utils.data.Dataset`.

Implement `MarsTorchDataset` that pytorch can fetch data directly from Mars cluster.

User code will be like:
``` Python
import torch.nn as nn
import torch.distributed as dist
import torch.optim as optim
import torch.utils.data

import mars.tensor as mt

data = mt.named_tensor('data')
labels = mt.named_tensor('labels')
train_dataset = MarsTorchDataset(data, labels)

train_sampler = torch.utils.data.distributed.DistributedSampler(train_dataset)
train_loader = torch.utils.data.DataLoader(dataset=train_dataset,
                                           batch_size=32,
                                           shuffle=False,
                                           sampler=train_sampler)

model = nn.parallel.DistributedDataParallel(get_model())
optimizer = optim.SGD(model.parameters(),
                      lr=0.01, momentum=0.5)
criterion = nn.BCELoss()

for _ in range(2):
    # 2 epochs
    for _, (batch_data, batch_labels) in enumerate(train_loader):
        outputs = model(batch_data)
        loss = criterion(outputs.squeeze(), batch_labels)
        optimizer.zero_grad()
        loss.backward()
        optimizer.step()
```

## Related issue number

Resolves #855 .

### CODE REVIEW

File: mars/scheduler/session.py [hunk 1]
```diff
@@ -221,10 +231,13 @@ def get_sessions(self):
     def create_session(self, session_id, **kwargs):
         uid = SessionActor.gen_uid(session_id)
         scheduler_address = self.get_scheduler(uid)
-        session_ref = self.ctx.create_actor(SessionActor, uid=uid, address=scheduler_address,
-                                            session_id=session_id, **kwargs)
-        self._session_refs[session_id] = session_ref
-        return session_ref
+        if session_id in self._session_refs:
```
Comment (qinxuye) [inline mars/scheduler/session.py]: Add a `if_not_exist` argument here.

File: mars/tensor/core.py [hunk 2]
```diff
@@ -780,5 +780,19 @@ def mutable_tensor(name, shape=None, dtype=np.float_, fill_value=None, chunk_siz
                                              fill_value=fill_value, chunk_size=chunk_size)
 
 
+def named_tensor(name, shape=None):
```
Comment (qinxuye) [inline mars/tensor/core.py]: Let `MarsDataset` support pass a name. Hide `named_tensor` for now until context can be used everywhere.

File: mars/tensor/__init__.py [hunk 3]
```diff
@@ -56,7 +56,7 @@
 from .einsum import einsum
 # noinspection PyUnresolvedReferences
 from .lib.index_tricks import mgrid, ogrid, ndindex
-from .core import mutable_tensor
+from .core import mutable_tensor, named_tensor
```
Comment (qinxuye) [inline mars/tensor/__init__.py]: Do not expose `named_tensor` for now.

File: mars/worker/transfer.py [hunk 4]
```diff
@@ -783,8 +783,19 @@ def pre_destroy(self):
         self._result_copy_ref.destroy()
         super().pre_destroy()
 
-    def fetch_data(self, session_id, chunk_key, index_obj=None):
-        compression_type = dataserializer.CompressType(options.worker.transfer_compression)
+    def fetch_batch_data(self, session_id, chunk_keys, index_objs=None, compression_type=None):
+        results = []
+        if index_objs is not None:
+            for chunk_key, index_obj in zip(chunk_keys, index_objs):
```
Comment (qinxuye) [inline mars/worker/transfer.py]: Logic could be simplified.
